tests/ParserSpec.hs

```diff
@@ -77,7 +78,23 @@ parserSpec = do
       -- Do we still correctly parse merge windows after a deploy environment
       it "can parse a merge window after an explicit environment" $
         dummyParse "@bot merge and deploy to production on friday" `shouldBe`
-          Success (Approve $ MergeAndDeploy (DeployEnvironment "production"), OnFriday)
+          Success (Approve $ MergeAndDeploy AllSubprojects (DeployEnvironment "production"), OnFriday)
+
+      it "allows a specific subproject to be deployed" $
+        dummyParse "@bot merge and deploy aaa to production" `shouldBe`
+          Success (Approve $ MergeAndDeploy (OnlySubprojects ["aaa"]) (DeployEnvironment "production"), AnyDay)
+
+      it "allows many specific subprojects to be deployed" $
+        dummyParse "@bot merge and deploy aaa, bbb to production" `shouldBe`
+          Success (Approve $ MergeAndDeploy (OnlySubprojects ["aaa", "bbb"]) (DeployEnvironment "production"), AnyDay)
+
+      it "fails when an unknown subproject is specified" $
+        dummyParse "@bot merge and deploy ccc to production" `shouldBe`
+          -- I'm not super happy with this error message, it _should_ say that we expect to see
+          -- either 'to', or the name of a known subproject.
+          ParseError
+            "comment:1:23:\n  |\n1 | @bot merge and deploy ccc to production\n\
+            \  |                       ^^\nunexpected \"cc\"\nexpecting \"to\" or white space\n"
```

Comment:
Can you add a test for `@bot merge and deploy aaa` without the environment, to see what it does.

Reply:
Two tests added:

  - That we can use subprojects with an implicit environment (`@bot merge and deploy aaa`)
  - That we can do the above with a merge window (`@bot merge and deploy aaa on friday`)